maintainers/scripts/regenerate-hackage-packages.sh

```diff
@@ -0,0 +1,12 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i bash -p coreutils
+set -euxo pipefail
```

Comment:
The message body of the PR says that this still needs documentation, and I agree.

It would be nice to have a couple sentences at the top of this script (and `update-hackage-packages.sh`)  describing what it does, why it is used, who normally uses it, etc.